server/py/services/api/api/endpoints/pipelines.py

```diff
@@ -145,23 +151,125 @@ async def retry_pipeline(
     auth_info: mlrun.common.schemas.AuthInfo = fastapi.Depends(
         framework.api.deps.authenticate_request
     ),
+    submit_mode: str = fastapi.Query(
```

Comment:
This method is becoming quite long, maybe can be split a bit?
You already have `services.api.crud.Pipelines().retry_pipeline`. Maybe rename it to `services.api.crud.Pipelines().rerun_pipeline_direct` and add `services.api.crud.Pipelines().rerun_pipeline` with the rest of the code